uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -127,13 +126,18 @@ define([
         /**
          * Gets the current balance change according to a current work pattern
          *
+         * @param  {String} calculationUnit (days|hours)
          * @return {Promise} resolves to an object containing
          *   a balance change amount and a detailed breakdown
          */
-        calculateBalanceChange: function () {
-          return LeaveRequestAPI.calculateBalanceChange(
-            _.pick(this, ['contact_id', 'from_date',
-              'from_date_type', 'to_date', 'to_date_type']));
+        calculateBalanceChange: function (calculationUnit) {
+          var params = ['contact_id', 'from_date', 'to_date', 'type_id', 'from_date_type', 'to_date_type'];
+
+          if (calculationUnit === 'hours') {
+            _.pull(params, 'from_date_type', 'to_date_type');
```

Comment:
The reason I did pull instead of push is that at the moment calculateBalanceChange does not accept params. I wanted to leave it like that and default to "days" and only if a param is passed - check.